api/renderers.py

```diff
@@ -46,11 +50,199 @@ def extract_attributes(fields, resource):
+                if not resolved:
+                    continue
+
+                if isinstance(field.child_relation, ResourceRelatedField):
+                    # special case for ResourceRelatedField
+                    relation_data = {
+                        'data': cls.encode_ids(resource.get(field_name))
+                    }
+
+                    field_links = field.child_relation.get_links(resource_instance)
+                    relation_data.update(
+                        {'links': field_links}
+                        if field_links else dict()
+                    )
+                    relation_data.update(
+                        {
+                            'meta': {
+                                'count': len(resource.get(field_name))
+                            }
+                        }
+                    )
+                    data.update({field_name: relation_data})
+                    continue
+
+                relation_data = list()
+                for nested_resource_instance in relation_instance:
+                    nested_resource_instance_type = (
+                        relation_type or
+                        utils.get_resource_type_from_instance(nested_resource_instance)
+                    )
+
+                    relation_data.append(OrderedDict([
+                        ('type', nested_resource_instance_type),
+                        ('id', cls.encode_id(nested_resource_instance.pk, nested_resource_instance_type))
+                    ]))
+                data.update({
+                    field_name: {
+                        'data': relation_data,
+                        'meta': {
+                            'count': len(relation_data)
+                        }
+                    }
+                })
+                continue
+
+            if isinstance(field, ListSerializer):
+                resolved, relation_instance = utils.get_relation_instance(resource_instance, source, field.parent)
+                if not resolved:
+                    continue
+
+                relation_data = list()
+
+                serializer_data = resource.get(field_name)
+                resource_instance_queryset = list(relation_instance)
+                if isinstance(serializer_data, list):
+                    for position in range(len(serializer_data)):
+                        nested_resource_instance = resource_instance_queryset[position]
+                        nested_resource_instance_type = (
+                            relation_type or
+                            utils.get_resource_type_from_instance(nested_resource_instance)
+                        )
+
+                        relation_data.append(OrderedDict([
+                            ('type', nested_resource_instance_type),
+                            ('id', cls.encode_id(nested_resource_instance.pk, nested_resource_instance_type))
```

Comment:
If `nested_resource_instance` is just a model instance `IDObfuscator.encode(nested_resource_instance)` will work.